hw/dv/sv/pwm_agent/pwm_agent.core

```diff
@@ -10,13 +10,14 @@ filesets:
       - lowrisc:dv:dv_utils
       - lowrisc:dv:dv_lib
     files:
-      - pwm_if.sv
       - pwm_agent_pkg.sv
+      - pwm_if.sv
       - pwm_item.sv: {is_include_file: true}
       - pwm_agent_cfg.sv: {is_include_file: true}
       - pwm_agent_cov.sv: {is_include_file: true}
       - pwm_driver.sv: {is_include_file: true}
       - pwm_monitor.sv: {is_include_file: true}
+      - pwm_sequencer.sv: {is_include_file: true}
       - pwm_agent.sv: {is_include_file: true}
```

Comment:
PWM is a monitor-only agent. There is no usecase for it to have driving capability. Please remove the following:
agent, driver, sequencer, item and cov. 

Just keep the `pwm_if`, `pwm_monitor` and `pwm_cfg`. Instantiate the monitor directly in the env, instead of instantiating the agent. Agent really is a wrapper that provides run-time active/passive mode / driving direction switching capability, none of which is needed here. 

You can implement coverage in the monitor itself - the monitor code is not large. 
You can replace the `pwm_item` with a `struct` to keep it simpler. Example: `typedef struct { int en_cycles; int duty_cycle; } pwm_item_t;`. 
Please parameterize the agent classes with `NUM_PWM_CHANNELS`. Agents should not reference parameters from a package. They should be parameterized instead.